frontend/src/components/LineChart.vue

```diff
@@ -156,8 +169,29 @@ const options = {
   parsing: getParsing,
   plugins: {
     legend: {
-      display: false,
-      position: 'bottom'
+      display: true,
+      position: 'top',
+      align : 'end',
+      labels: {
+        usePointStyle: true,
+        generateLabels: () => chartStore.generateDataQualityLegend(),
+        font: {
+          size: 12,
+        },
+        boxWidth: 20,
+        padding: 10,
+      },
+      title: {
+        display: true,
+        text: 'Data Quality',
+        font: {
+          size: 16,
+          weight: 'bold',
+        },
+        padding: {
+          top: 10,
+        },
+      },
```

Comment:
Nice!
Also add to `NodeChart.vue`?

Reply:
(in other words -- you added the legend to the Time charts but not to the Distance charts)

Reply:
Great suggestion! Done!